File: apps/ios/OpenFrame/Model/Kiosk.swift
import Foundation

struct Kiosk: Identifiable, Codable, Sendable, Equatable {
    let id: String
    let name: String
    let isActive: Bool
    let displayMode: String?
    let displayType: String?
    let colorScheme: String?
    let screensaverEnabled: Bool
    let dashboards: [KioskDashboard]
    let lastAccessedAt: String?
}

struct KioskDashboard: Identifiable, Codable, Sendable, Equatable {
    let id: String
    let type: String
    let name: String
    let icon: String?
    let pinned: Bool
}

struct KioskSavedFile: Identifiable, Codable, Sendable, Equatable {
    let id: String
    let name: String
    let fileType: String
    let mimeType: String?
    let pageCount: Int?
    let fileSize: Int?
    let createdAt: String?
}

File: apps/android/app/src/main/java/us/openframe/app/domain/model/Kiosk.kt
package us.openframe.app.domain.model

data class Kiosk(
    val id: String,
    val name: String,
    val isActive: Boolean,
    val displayMode: String?,
    val displayType: String?,
    val colorScheme: String?,
    val screensaverEnabled: Boolean,
    val dashboards: List<KioskDashboard>,
    val lastAccessedAt: String?,
)

data class KioskDashboard(
    val id: String,
    val type: String,
    val name: String,
    val icon: String?,
    val pinned: Boolean,
)
